src/radiation/radiation_system.hpp

```diff
@@ -2006,4 +1934,524 @@ void RadSystem<problem_t>::AddSourceTerms(array_t &consVar, arrayconst_t &radEne
 	});
 }
 
+template <typename problem_t>
+void RadSystem<problem_t>::AddSourceTermsSingleGroup(array_t &consVar, arrayconst_t &radEnergySource, amrex::Box const &indexRange, Real dt_radiation,
```

Comment:
The source term implementation has become very complicated. Maybe this function can go in a separate header file?

Reply:
Maybe in a future PR?

Reply:
Is there a reason not to do it in this PR? It should take about 60 seconds to implement.

Reply:
In this case, the right way to do it is probably with a recursive include. That is, at the end of `radiation_system.hpp`, you can do
```
#include "AddSourceTerms.hpp  // IWYU pragma: export
#include "AddSourceTermsMultigroup.hpp  // IWYU pragma: export
```
and then in each of those header files, just cut-and-paste the functions, and add this at the top of each file:
```
// IWYU pragma: private; include "radiation_system.h"
```
(See https://clangd.llvm.org/guides/include-cleaner#iwyu-pragmas)

Reply:
OK, this looks good. I've updated it. The code compiles and runs without any problem. The only drawback I'm experiencing is that vscode won't do variable indexing inside add_source_terms_multi_group.hpp, but I think this can be fixed.